agent/api_manager/osv_service_api.py

```diff
@@ -0,0 +1,175 @@
+"""This module provides utility functions to query the OSV API for vulnerability information
+related to a specific package version."""
+import dataclasses
+import json
+import logging
+from typing import Iterator, Any
+
+import requests
+import tenacity
+from ostorlab.agent.kb import kb
+from ostorlab.agent.mixins import agent_report_vulnerability_mixin
+
+from agent import cve_service_api
+from agent import osv_output_handler
+
+logger = logging.getLogger(__name__)
+
+OSV_ENDPOINT = "https://api.osv.dev/v1/query"
+NUMBER_RETRIES = 3
+WAIT_BETWEEN_RETRIES = 2
+
+
+@dataclasses.dataclass
+class VulnData:
+    risk: str
+    description: str
+    summary: str
+    fixed_version: str | None
+    cvss_v3_vector: str | None
+    references: list[dict[str, str]]
+    cves: list[str]
+
+
+@tenacity.retry(
+    stop=tenacity.stop_after_attempt(NUMBER_RETRIES),
+    wait=tenacity.wait_fixed(WAIT_BETWEEN_RETRIES),
+    retry=tenacity.retry_if_exception_type(),
+    retry_error_callback=lambda retry_state: retry_state.outcome.result()
+    if retry_state.outcome is not None
+    else None,
+)
+def query_osv_api(
+    package_name: str, version: str, ecosystem: str
+) -> dict[str, Any] | None:
+    """Query the OSV API with the specified version, package name, and ecosystem.
+    Args:
+        version: The version to query.
+        package_name: The name of the package to query.
+        ecosystem: The ecosystem of the package e.g., javascript.
+    Returns:
+        The API response text if successful, None otherwise.
+    """
+    data = {
+        "version": version,
+        "package": {"name": package_name, "ecosystem": ecosystem},
+    }
+    headers = {"Content-Type": "application/json"}
+    response = requests.post(OSV_ENDPOINT, data=json.dumps(data), headers=headers)
+
+    if response.status_code == 200:
+        resp: dict[str, Any] = response.json()
+        return resp
+
+    return None
+
+
+def parse_output(
+    api_response: dict[str, Any], api_key: str | None = None
+) -> list[VulnData]:
+    """Parse the OSV API response to extract vulnerabilities.
+    Args:
+        api_response: The API response json.
+        api_key: The API key.
+    Returns:
+        Parsed output.
+    """
+    try:
+        vulnerabilities = api_response.get("vulns", [])
+
+        parsed_vulns = []
+        for vulnerability in vulnerabilities:
+            risk = vulnerability.get("database_specific", {}).get("severity")
+            filtered_cves = [
+                alias for alias in vulnerability.get("aliases", []) if "CVE" in alias
+            ]
+            if risk is None:
+                risk_ratings = []
+                for cve in filtered_cves:
+                    cve_data = cve_service_api.get_cve_data_from_api(cve, api_key)
+                    risk_ratings.append(cve_data.risk)
+                risk = osv_output_handler.calculate_risk_rating(risk_ratings)
+            elif risk == "MODERATE":
+                risk = "MEDIUM"
```

Comment:
Why is there only moderate. How about the other values?

Reply:
for others (high, etc.) we use them as they are because they match the severity levels we use, but for moderate its not the case: 
![image](https://github.com/Ostorlab/agent_osv/assets/144013278/014fc5d8-ceeb-46fb-8ae2-5aae99a0149e)
![image](https://github.com/Ostorlab/agent_osv/assets/144013278/a4dbc27d-cd8e-4fad-9f75-b3e096e8148d)
![image](https://github.com/Ostorlab/agent_osv/assets/144013278/88039b69-d365-4b3b-b2e1-c0352e14e90b)
![image](https://github.com/Ostorlab/agent_osv/assets/144013278/fc8bb328-8177-48fb-9439-f0430488ccfe)